### PR TITLE
Un-generic some methods

### PR DESCRIPTION
**Pull request by lingyanhao**
In particular, the `getItemList`, `setItem`, `hasItem`, `deleteItem`, `addItem` methods

### CODE REVIEW

File: src/test/java/seedu/address/model/RestaurantBookTest.java [hunk 1]
```diff
@@ -114,22 +116,29 @@ public void removeListener_withInvalidationListener_listenerRemoved() {
      */
     private static class RestaurantBookStub implements ReadOnlyRestaurantBook {
         private final ObservableList<Member> members = FXCollections.observableArrayList();
-        private final ObservableList<Booking> bookings = FXCollections.observableArrayList();
```
Comment (limli) [inline src/test/java/seedu/address/model/RestaurantBookTest.java:117]: why is this deleted?

Comment (lingyanhao) [reply src/test/java/seedu/address/model/RestaurantBookTest.java:117]: As you can see in this test, only member-level operations are tested. In the event that we need this method, we can always put it back.

As of now, this list does not serve any purpose.

Comment (limli) [reply src/test/java/seedu/address/model/RestaurantBookTest.java:117]: ok, but later should test all 4

**Review by limli — COMMENTED** [review]
looks good. Still have some stuff to un generic, to be done in another PR.
